doc/specs/#607 - Commandline Arguments for the Windows Terminal.md

```diff
@@ -0,0 +1,534 @@
+intended for the given window. So the first `new-tab` in the file will create
+the window, and all subsequent `new-tab` commands will create tabs in that same
+window.
+ -->
+
+TODO: What should we do with an entirely empty command? My gut says _ignore it_.
+For example, `wt ; ; ; ` should not just open 4 tabs. But then what about `wt`
+by itself? We want that to open a new tab with the default profile. So maybe we
+should allow them? Or allow them on the first command only?
+
+
+### Options
+
+#### `--help,-h,-?`
+Runs the `help` command.
+
+#### `--version,-v`
+Runs the `version` command.
+
+#### `--session,-s session-id`
+Run these commands in the given Windows Terminal session. Enables opening new
+tabs in already running Windows Terminal windows. This feature is dependent upon
+other planned work landing, so is only provided as an example, of what it might
+look like. See [Future Considerations](#Future-Considerations) for more details.
+
+#### `--file,-f congifuration-file`
+Run these commands in the given Windows Terminal session. Enables opening new
+tabs in already running Windows Terminal windows. See [Future
+Considerations](#Future-Considerations) for more details.
+
+### Commands
+
+#### `help`
+
+`help`
+
+Display the help message
+
+#### `version`
+
+`version`
+
+Display version info for the Windows Terminal
+
+#### `open-settings`
+
+`open-settings [--defaults,-d]`
+
+Open the settings file.
+
+**Parameters**:
+* `--defaults,-d`: Open the `defaults.json` file instead of the `profiles.json`
+  file.
+
+#### `list-profiles`
+
+`list-profiles [--all,-A] [--showGuids,-g]`
+
+Displays a list of each of the available profiles. Each profile displays it's
+name, seperated by newlines.
+
+**Parameters**:
+* `--all,-A`: Show all profiles, including profiles marked `"hidden": true`.
+* `--showGuids,-g`:  In addition to showing names, also list each profile's
+  guid. These GUIDs should probably be listed _first_ on each line, to make
+  parsing output easier.
+
+#### `new-tab`
+
+`new-tab [--initialPosition x,y]|[--maximized]|[--fullscreen] [--initialRows rows] [--initialCols cols] [terminal_parameters]`
+
+Opens a new tab with the given customizations. On it's _first_ invocation, also
+opens a new window. Subsequent `new-tab` commands will all open new tabs in the
+same window.
+
+**Parameters**:
+* `--initialPosition x,y`: Create the new Windows Terminal window at the given
+  location on the screen in pixels. This parameter is only used when initially
+  creating the window, and ignored for subsequent `new-tab` commands. When
+  combined with any of `--maximized` or `--fullscreen`, TODO: what do?
+* `--initialRows rows`: Create the terminal window with `rows` rows (in
+  characters). If omitted, uses the value from the user's settings. This
+  parameter is only used when initially creating the window, and ignored for
+  subsequent `new-tab` commands. When combined with any of `--maximized` or
+  `--fullscreen`, TODO: what do?

```

Comment:
I'd say ignore initialPosition, initialRows, and initialCols if `--maximized` or `--fullscreen` is specified.

Reply:
That's a good point, but what about `--maximized` _and_ `--fullscreen`? Biggest setting wins? i.e. `wt --minimized --maximized --fullscreen` opens _fullscreen_?

Reply:
🤔 I kinda wanna just throw an error in those cases where you really should only have one specified. Not sure how to go about that situation.

Also, to go back to my earlier comment, is it possible to honor initialPosition, initialRows, and initialCols but still have one of minimized, maximized, or fullscreen as well?

I'm thinking of the behavior where you specify `minimized`, but you want your terminal window to be 120x60 on your second screen when you un-minimize/un-fullscreen/un-maximize.

Reply:
Hmm. That might be feasible, though it'd certainly be a little wonky to plumb all that through in just the right way.

Reply:
biggest one wins.

Reply:
or invalid param

Reply:
Hey so CLI11 actually lets us specify when args should and shouldn't be combined, so we can make `--maximized` and `--fullscreen` exclusive, and have it format a message suggesting that the user shouldn't combine the two for us. In light of that, I'm leaning towards _error message_